cmd/routed-eni-cni-plugin/driver/driver.go

```diff
@@ -267,354 +271,358 @@ func WaitForAddressesToBeStable(ifName string, timeout time.Duration) error {
 	}
 }
 
-// SetupNS wires up linux networking for a pod's network
-func (os *linuxNetwork) SetupNS(hostVethName string, contVethName string, netnsPath string, v4Addr *net.IPNet, v6Addr *net.IPNet,
-	deviceNumber int, vpcCIDRs []string, useExternalSNAT bool, mtu int, log logger.Logger) error {
-	log.Debugf("SetupNS: hostVethName=%s, contVethName=%s, netnsPath=%s, deviceNumber=%d, mtu=%d", hostVethName, contVethName, netnsPath, deviceNumber, mtu)
-	return setupNS(hostVethName, contVethName, netnsPath, v4Addr, v6Addr, deviceNumber, vpcCIDRs, useExternalSNAT, os.netLink, os.ns, mtu, log, os.procSys)
-}
-
-func setupNS(hostVethName string, contVethName string, netnsPath string, v4Addr *net.IPNet, v6Addr *net.IPNet, deviceNumber int, vpcCIDRs []string, useExternalSNAT bool,
-	netLink netlinkwrapper.NetLink, ns nswrapper.NS, mtu int, log logger.Logger, procSys procsyswrapper.ProcSys) error {
+// SetupPodNetwork wires up linux networking for a pod's network
+func (n *linuxNetwork) SetupPodNetwork(hostVethName string, contVethName string, netnsPath string, v4Addr *net.IPNet, v6Addr *net.IPNet,
+	deviceNumber int, mtu int, log logger.Logger) error {
+	log.Debugf("SetupPodNetwork: hostVethName=%s, contVethName=%s, netnsPath=%s, v4Addr=%v, v6Addr=%v, deviceNumber=%d, mtu=%d",
+		hostVethName, contVethName, netnsPath, v4Addr, v6Addr, deviceNumber, mtu)
 
-	hostVeth, err := setupVeth(hostVethName, contVethName, netnsPath, v4Addr, v6Addr, netLink, ns, mtu, procSys, log)
+	hostVeth, err := n.setupVeth(hostVethName, contVethName, netnsPath, v4Addr, v6Addr, mtu, log)
 	if err != nil {
-		return errors.Wrapf(err, "setupNS network: failed to setup veth pair.")
+		return errors.Wrapf(err, "SetupPodNetwork: failed to setup veth pair")
 	}
 
-	log.Debugf("Setup host route outgoing hostVeth, LinkIndex %d", hostVeth.Attrs().Index)
+	var containerAddr *net.IPNet
+	if v4Addr != nil {
```

Comment:
`v4Addr.IP.To4()` and `v6Addr.IP.To16()` are helpful to have here. For ex, if an IPv6 address ended being passed as `v4Addr` here - these checks can help catch them. `To4` will be helpful in particular to identify these mismatches.

Reply:
i believe the old code's To4 and To16 never triggers since the `addrHostAddr` will be nil and trigger nil-pointer exception if To4 and To16 fails. 
I removed to be align with our original PodENI code as well. 
```
var addrHostAddr *net.IPNet
	//We only support either v4 or v6 modes.
	if v4Addr != nil && v4Addr.IP.To4() != nil {
		addrHostAddr = &net.IPNet{
			IP:   v4Addr.IP,
			Mask: net.CIDRMask(32, 32)}
	} else if v6Addr != nil && v6Addr.IP.To16() != nil {
		addrHostAddr = &net.IPNet{
			IP:   v6Addr.IP,
			Mask: net.CIDRMask(128, 128)}
	}
```

Reply:
So, it helps us catch if an incorrect address family IP address is passed in via v4Addr (or) v6Addr. Is the concern about a `nil` exception if these checks fail? I mean incorrect IP addresses being passed in here is an issue to begin with anyways. We can probably address the `nil` pointer concern instead.

Reply:
chatted offline, we'll add a comment to assert that we expect v4/v6 to have correct IPFamily